### PR TITLE
Implement Swoosh.Adapters.MsGraph

### PR DESCRIPTION
**Pull request by justindotpub**
As discussed in https://github.com/swoosh/swoosh/issues/744.

### CODE REVIEW

File: test/support/conn_parser.ex [hunk 1]
```diff
@@ -8,7 +8,7 @@ defmodule Swoosh.ConnParser do
         Plug.Parsers.JSON
       ])
       |> Keyword.put_new(:json_decoder, Swoosh.json_library())
-      |> Keyword.put_new(:pass, ["message/rfc822"])
+      |> Keyword.put_new(:pass, ["message/rfc822", "text/plain"])
```
Comment (justindotpub) [inline test/support/conn_parser.ex:11]: Without this, `conn = parse(conn)` in tests would blow up.

**Comment by princemaple** [comment]
One last thing, is plug needed for this adapter? I don't see it being used other than in the tests.

**Comment by justindotpub** [comment]
Good catch. I didn’t use it explicitly but assumed it was some requirement under the covers, perhaps due to the use of the api client or something. Let me know I need to make that change. I’ll assume you are otherwise, so I don’t mess up what you’re doing.

**Comment by justindotpub** [comment]
I confirmed, the mailgun adapter that I used as my starting point explicitly used `Plug.Conn.Query.encode`.  I don't need Plug.

**Comment by princemaple** [comment]
Thank you @justindotpub ! This is a great PR.

**Comment by justindotpub** [comment]
Thank you for the support @princemaple! ❤️

**Pull request merged**